Sources/App/Controllers/ExploreController.swift

```diff
@@ -55,12 +60,17 @@ final class ExploreController {
         let search = request.query?["term"]?.string ?? ""
 
         // Only search through accessory and manufacturer name.
-        let categories = try Category.all()
         let accessories = try Accessory.makeQuery().filter("approved", true).all().sorted(by: { $0.date.compare($1.date) == .orderedDescending }).filter { accessory -> Bool in
             let manufacturerResult = try accessory.manufacturer.get()?.name.lowercased().contains(search) ?? false
             let nameResult = accessory.name.lowercased().contains(search)
             return manufacturerResult || nameResult
         }
+
+        let categories = try Category.all()
+
+        let accessoryCount = categories.reduce(0, {$0 + $1.accessoriesCount})
```

Comment:
In this case you can see that at line 63 we still query all the accessories and then we sort them. To avoid doing the `reduce` on all categories here, you could first store the result of `try Accessory.makeQuery().filter("approved", true).all()` in a variable, store the count and then do the `sort` and `filter`.

Reply:
Yup that make sense.  I'll update that handler.